File: libartivity/src/ObjectModel/Entities/InformationElement.cpp
#include "InformationElement.h"
#include "FileDataObject.h"

using namespace artivity;

void InformationElement::setStoredAs(FileDataObjectRef fdo)
{
    _fileDataObject = fdo;
    addProperty(nie::isStoredAs, fdo);
}

FileDataObjectRef InformationElement::getStoredAs()
{
    return _fileDataObject;
}

File: Artivity.DataModel/ObjectModel/Entities/InformationElement.cs
using Semiodesk.Trinity;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace Artivity.DataModel
{
    [RdfClass(NIE.InformationElement)]
    public class InformationElement : Entity
    {
        public InformationElement(Uri uri) : base(uri) { }
        public InformationElement(string uri) : base(uri) { }

        [RdfProperty(NIE.isStoredAs)]
        public FileDataObject DataObject { get; set; }
    }
}
